agent/lib/kontena/helpers/weave_helper.rb

```diff
@@ -1,15 +1,32 @@
 require 'docker'
 require_relative 'iface_helper'
+require_relative 'wait_helper'
+
 
 module Kontena
   module Helpers
     module WeaveHelper
+      include WaitHelper
+
+      def network_adapter
+        Celluloid::Actor[:network_adapter]
+      end
+
+      def wait_weave_running?
+        wait!(300, "waiting for weave running") {
+          network_adapter.running?
+        }
+      end
+
+      def wait_network_ready?
+        wait!(300, "waiting for all network components running") {
+          network_adapter.network_ready?
+        }
+      end
 
-      # @return [Boolean]
-      def weave_running?
-        weave = Docker::Container.get('weave') rescue nil
-        return false if weave.nil?
-        weave.info['State']['Running'] == true
+      def weave_api_ready?
```

Comment:
Does this throw an exception if weave is not responding (SocketError or similar)?